common/src/main/java/us/myles/ViaVersion/api/PacketWrapper.java

```diff
@@ -488,7 +490,10 @@ public UserConnection user() {
      */
     public void resetReader() {
         // Move all packet values to the readable for next packet.
-        this.readableObjects.addAll(0, packetValues);
+        Collections.reverse(packetValues);
+        for (Pair<Type, Object> obj : this.packetValues) {
+            this.readableObjects.addFirst(obj);
+        }
         this.packetValues.clear();
```

Comment:
for (int i = packetValues.size() - 1; i >= 0; i--) {
            this.readableObjects.addFirst(this.packetValues.get(i));
        }
        this.packetValues.clear();